app/Http/Controllers/AlbumController.php

```diff
@@ -208,6 +211,32 @@ public function setLicense(SetAlbumLicenseRequest $request): void
 		$request->album()->save();
 	}
 
+	/**
+	 * Upload a track for the Album.
+	 *
+	 * @param SetAlbumTrackRequest $request
+	 * @param Tracks               $tracks
+	 *
+	 * @return void
+	 */
+	public function setTrack(SetAlbumTrackRequest $request, Tracks $tracks): void
+	{
+		$tracks->set($request->album()->id, $request->file);
```

Comment:
I just realize that you already have implemented a specific request class :+1: :tada: 

But if you already have the album as part of the request, it is not necessary to get the ID of the album, just to fetch the same album from the DB inside the action again. This is an unnessary DB roundtrip.

You can simply do something like that

```php
$request->album()->track_short path = ...
```